packages/amplify-graphql-model-transformer/src/resources/custom-resource-lambda/custom-resource-handler.ts

```diff
@@ -0,0 +1,440 @@
+        await retry(
+          async () => await doesTableExist(event.PhysicalResourceId!),
+          (res) => res === false,
+        );
+        console.log(`Table '${event.PhysicalResourceId}' does not exist. Deletion is finished.`);
+
+        const createTableInput = toCreateTableInput(tableDef);
+        const response = await createNewTable(createTableInput);
+        result = {
+          PhysicalResourceId: response.tableName,
+          Data: {
+            TableArn: response.tableArn,
+            TableStreamArn: response.streamArn,
+            TableName: response.tableName,
+          },
+        };
+        log('Returning result', result);
+        return result;
+      }
+      // When normal updates happen
+      const nextGsiUpdate = getNextGSIUpdate(describeTableResult.Table, tableDef);
+      log('Computed next update', nextGsiUpdate);
+      if (!nextGsiUpdate) {
+        result = {
+          PhysicalResourceId: event.PhysicalResourceId,
+          Data: {
+            TableArn: describeTableResult.Table.TableArn,
+            TableStreamArn: describeTableResult.Table.LatestStreamArn,
+            TableName: describeTableResult.Table.TableName,
+          },
+        };
+        return result; // nothing to update
+      }
+
+      // TODO: merge gsi update with other table updates
+
+      const updateTableInput: UpdateTableInput = nextGsiUpdate;
+      log('Merged gsi update with other table updates', updateTableInput);
+
+      console.log('Initiating table update');
+      await ddbClient.updateTable(updateTableInput).promise();
+      result = {
+        PhysicalResourceId: event.PhysicalResourceId,
+        Data: {
+          TableArn: describeTableResult.Table.TableArn,
+          TableStreamArn: describeTableResult.Table.LatestStreamArn,
+          TableName: describeTableResult.Table.TableName,
+        },
+      };
+      return result;
+    case 'Delete':
+      console.log('Initiating table deletion');
+      if (!event.PhysicalResourceId) {
+        throw new Error(`Could not find the physical ID for the resource`);
+      }
+      try {
+        await ddbClient.deleteTable({ TableName: event.PhysicalResourceId }).promise();
+      } catch (err) {
+        // TODO only swallow NotExist errors
+      }
+  }
+  // after this function exits, the state machine will invoke isComplete in a loop until it returns finished or the state machine times out
+};
+
+export const isComplete = async (
+  event: AWSCDKAsyncCustomResource.IsCompleteRequest,
+): Promise<AWSCDKAsyncCustomResource.IsCompleteResponse> => {
+  log('got event', event);
+  if (event.RequestType === 'Delete') {
+    // nothing else to do on delete
+    console.log('Delete is finished');
+    return finished;
+  }
+  if (!event.PhysicalResourceId) {
+    throw new Error('PhysicalResourceId not set in call to isComplete');
+  }
+  console.log('Fetching current table state');
+  const describeTableResult = await retry(
```

Comment:
Similar to above comment: we retry the describeTable here, but don't above when we check for existing state. Why the discrepancy?

Reply:
This `describe` is inside `isComplete`. The `retry` here is to avoid the delay of `Create` event not already populating the new table. 

For the `check for existing state` mentioned, it is in the `onEvent` handler during an `Update` event. The retry is not used but instead a error will be thrown if no table returned in response, which means the table does not exist or the physical Id is not correct, in which case the retry is not needed.